sw/device/tests/rstmgr_cpu_info_test.c

```diff
@@ -0,0 +1,202 @@
+  temp_dump[kCpuDumpIdxPreviousValid] = 1;
+
+  CHECK(flash_ctrl_testutils_write(&flash_ctrl, addr, 0, temp_dump,
+                                   kDifFlashCtrlPartitionTypeData, 9) == 0);
+
+  read_error();
+}
+
+bool test_main(void) {
+  dif_rstmgr_t rstmgr;
+  dif_aon_timer_t aon_timer;
+  dif_pwrmgr_t pwrmgr;
+
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  // Initialize flash_ctrl
+  CHECK_DIF_OK(dif_flash_ctrl_init_state(
+      &flash_ctrl,
+      mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)));
+
+  // Enable flash access
+  flash_ctrl_testutils_default_region_access(&flash_ctrl,
+                                             /*rd_en*/ true,
+                                             /*prog_en*/ true,
+                                             /*erase_en*/ true,
+                                             /*scramble_en*/ false,
+                                             /*ecc_en*/ false,
+                                             /*he_en*/ false);
+
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+
+  if (rst_info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Booting for the first time, setting wdog");
+
+    global_error_cnt = 0;
+    uint32_t bark_cycles = aon_timer_testutils_get_aon_cycles_from_us(200);
+    uint32_t bite_cycles = aon_timer_testutils_get_aon_cycles_from_us(200);
+
+    // Set wdog as a reset source.
+    CHECK_DIF_OK(dif_pwrmgr_set_request_sources(&pwrmgr, kDifPwrmgrReqTypeReset,
+                                                kDifPwrmgrResetRequestSourceTwo,
+                                                kDifToggleEnabled));
+
+    // Setup the wdog bark and bite timeouts.
+    aon_timer_testutils_watchdog_config(&aon_timer, bark_cycles, bite_cycles,
+                                        false);
+
+    // Enable cpu info
+    CHECK_DIF_OK(dif_rstmgr_cpu_info_set_enabled(&rstmgr, kDifToggleEnabled));
+    read_error();
+  } else {
+    LOG_INFO("Comes back after bite");
+
+    size_t seg_size;
+
+    CHECK_DIF_OK(dif_rstmgr_cpu_info_dump_read(
+        &rstmgr, &dump[0], DIF_RSTMGR_CPU_INFO_MAX_SIZE, &seg_size));
+
+    for (int i = 0; i < seg_size; ++i) {
+      LOG_INFO("OBS_DUMP:%d: 0x%x", i, dump[i]);
+    }
+
+    uint32_t addr = (uint32_t)exp_dump;
+
+    for (int i = 0; i < seg_size; ++i) {
+      dif_rstmgr_cpu_info_dump_segment_t rdata = abs_mmio_read32(addr);
+
+      if (rdata == kSkipComp) {
+        LOG_INFO("CHK_DUMP:%d: %x : skipped", i, dump[i]);
+      } else {
+        CHECK(rdata == dump[i], "field mismatch: exp = 0x%x, obs = 0x%x", rdata,
+              dump[i]);
+      }
```

Comment:
you can probably remove the skip prints. (though I'm not sure I follow why you need to skip any comparisons, see above)

```suggestion
      if (rdata != kSkipComp)
        CHECK(rdata == dump[i], "field mismatch: exp = 0x%x, obs = 0x%x", rdata,
              dump[i]);
      }
```

Reply:
sure